intermediate_source/compiled_autograd_tutorial.rst

```diff
@@ -0,0 +1,301 @@
+Compiled Autograd: Capturing a larger backward graph for ``torch.compile``
+==========================================================================
+**Author:** `Simon Fan <https://github.com/xmfan>`_
+
+.. grid:: 2
+
+    .. grid-item-card:: :octicon:`mortar-board;1em;` What you will learn
+       :class-card: card-prerequisites
+
+       * How compiled autograd interacts with ``torch.compile``
+       * How to use the compiled autograd API
+       * How to inspect logs using ``TORCH_LOGS``
+
+    .. grid-item-card:: :octicon:`list-unordered;1em;` Prerequisites
+       :class-card: card-prerequisites
+
+       * PyTorch 2.4
+       * Complete the `Introduction to torch.compile <https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html>`_
+
+Overview
+--------
+Compiled Autograd is a ``torch.compile`` extension introduced in PyTorch 2.4
+that allows the capture of a larger backward graph.
+
+While ``torch.compile`` does capture the backward graph, it does so **partially**. The AOTAutograd component captures the backward graph ahead-of-time, with certain limitations:
+
+* Graph breaks in the forward lead to graph breaks in the backward
+* `Backward hooks <https://pytorch.org/docs/stable/notes/autograd.html#backward-hooks-execution>`_ are not captured
+
+Compiled Autograd addresses these limitations by directly integrating with the autograd engine, allowing
+it to capture the full backward graph at runtime. Models with these two characteristics should try
+Compiled Autograd, and potentially observe better performance.
+
+However, Compiled Autograd introduces its own limitations:
+
+* Added runtime overhead at the start of the backward for cache lookup
+* More prone to recompiles and graph breaks in dynamo due to the larger capture
+
+.. note:: Compiled Autograd is under active development and is not yet compatible with all existing PyTorch features. For the latest status on a particular feature, refer to `Compiled Autograd Landing Page <https://docs.google.com/document/d/11VucFBEewzqgkABIjebZIzMvrXr3BtcY1aGKpX61pJY>`_.
```

Comment:
I'm not sure why it's called Landing Page. Should it be called a Compiled Autograd Roadmap? Also, it feels like it would be better to point to issues on github with a specific label, perhaps, or a Github Project even?

Reply:
I'm expecting the content of this doc to change rapidly at the moment, so I'll keep it as a google doc for now. It's also something we did for dynamic shapes and custom operators.

The end goal once the development is more stable, is to move it into the docs